lib/govuk_message_queue_consumer/consumer.rb

```diff
@@ -14,17 +11,20 @@ class Consumer
 
     # Create a new consumer
     #
-    # @param queue_name [String] Your queue name. This is specific to your application.
     # @param exchange_name [String] Name of the exchange to bind to, for example `published_documents`
+    # @param queue_name [String] Your queue name. This is specific to your application,
+    #                            and should already exist and have a binding via puppet
     # @param processor [Object] An object that responds to `process`
-    # @param routing_key [String] The RabbitMQ routing key to bind the queue to
+    # @param bunny_connection [Object] A Bunny connection object derived from `Bunny.new`
     # @param statsd_client [Statsd] An instance of the Statsd class
-    def initialize(queue_name:, exchange_name:, processor:, routing_key: '#', statsd_client: NullStatsd.new)
-      @queue_name = queue_name
+    # @param logger [Object] A Logger object for emitting errors (to stderr by default)
+    def initialize(exchange_name:, queue_name:, processor:, bunny_connection:, statsd_client: NullStatsd.new, logger: Logger.new(STDERR))
       @exchange_name = exchange_name
+      @queue_name = queue_name
       @processor = processor
-      @routing_key = routing_key
+      @bunny_connection = bunny_connection
```

Comment:
Following on from https://github.com/alphagov/govuk_message_queue_consumer/pull/32/files#r60209085:

You could set `bunny_connection` to nil by default and then change this line into:

```ruby
@bunny_connection = bunny_connection || Bunny.new(RabbitMQConfig.new.from_environment)
```